perf_sim/config.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * config.cpp - implementation of Config class
+ * @author Pavel Kryukov pavel.kryukov@phystech.edu
+ * Copyright 2017 MIPT-MIPS
+ */
+
+/* Simulator modules */
+#include "config.h"
+
+/* Generic C++ */
+#include <iostream>
+
+/* boost - program option parsing */
+#include <boost/program_options.hpp>
+
+using namespace std;
+
+/* constructor and destructor */
+Config::Config() :
+    /* by default, there is no limit of simulation and silent mode is on */
+    num_steps( -1),
+    disassembly_on( false)
+{
+
+}
+
+Config::~Config()
+{
+
+}
+
+
+/* basic method */
+int Config::handleArgs( int argc, char** argv)
+{
+    namespace po = boost::program_options;
+
+    po::options_description description( "Allowed options");
+
+    description.add_options()
+        ( "binary,b",      po::value<string> ( &this->binary_filename),
+                                                            "Input binary file")
+
+        ( "numsteps,n",    po::value<int>    ( &this->num_steps),
+                               "Number of instructions to run. Defaults to -1,"
```

Comment:
Ok, just borrowed it from MDSP. What would be a good variant then to run simulation without limits?